src/runtime/manifest.ts

```diff
@@ -604,8 +604,8 @@ ${e.message}
               fields[name] = type;
               if (fields[name].refinement) {
                 fields[name].refinement = Refinement.fromAst(fields[name].refinement, {[name]: type.type});
-                typeData[name] = type.type;
               }
+              typeData[name] = type.type;
```

Comment:
Do we have a regression test for this? I didn't catch the bug in the previous review, so I think it would be pretty easy to reintroduce.

Reply:
Not an explicit one. But one of the new tests I wrote will implicitly fail if this bug exists (which is how I catch it in the first place).
Would you want me to add an explicit test case?

Reply:
Should be fine.